temoa/extensions/integer_capacity/components/integer_capacity.py

```
@@ -0,0 +1,101 @@
+    from temoa.types import ExprLike, Period, Region, Technology, Vintage
+
+
+def limit_integer_new_capacity_indices(
+    model: IntegerCapacityModel,
+) -> set[tuple[Region, Technology, Vintage]]:
+    indices = {
+        (r, t, v)
+        for r, t in model.limit_integer_new_capacity.sparse_keys()
+        for v in model.vintage_optimize
+    }
+    return indices
```

Ruff (0.15.20)

[warning] Unnecessary assignment to `indices` before `return` statement

Remove unnecessary assignment

(RET504)